src/EditorFeatures/CSharpTest/Completion/CompletionProviders/PropertySubPatternCompletionProviderTests.cs

```diff
@@ -0,0 +1,655 @@
+
+    void M()
+    {
+        _ = missing is { $$ }
+    }
+}
+";
+            await VerifyNoItemsExistAsync(markup);
+        }
+
+        [Fact]
+        public async Task PropertiesInRecursivePattern_MissingAfterColon()
+        {
+            var markup =
+@"
+class Program
+{
+    public int P1 { get; set; }
+    public int P2 { get; set; }
+
+    void M()
+    {
+        _ = this is { P1: $$ }
+    }
+}
+";
+            await VerifyNoItemsExistAsync(markup);
+        }
+
+        [Fact]
+        public async Task PropertiesInRecursivePattern_SecondProperty()
+        {
+            var markup =
+@"
+class Program
+{
+    public int P1 { get; set; }
+    public int P2 { get; set; }
+
+    void M()
+    {
+        _ = this is Program { P2: 1, $$ }
+    }
+}
+";
+            // VerifyItemExistsAsync also tests with the item typed.
+            await VerifyItemExistsAsync(markup, "P1");
+            await VerifyItemIsAbsentAsync(markup, "P2");
+        }
+
+        [Fact]
+        public async Task PropertiesInRecursivePattern_PositionalInFirstProperty()
+        {
+            var markup =
+@"
+class Program
+{
+    public D P1 { get; set; }
+
+    void M()
+    {
+        _ = this is Program { P1: ($$ }
+    }
+}
+class D
+{
+    public int P2 { get; set; }
+}
+";
+            await VerifyNoItemsExistAsync(markup);
+        }
+
+        [Fact]
+        public async Task PropertiesInRecursivePattern_PositionalInFirstProperty_AfterComma()
+        {
+            var markup =
+@"
+class Program
+{
+    public D P1 { get; set; }
+
+    void M()
+    {
+        _ = this is Program { P1: (1, $$ }
+    }
+}
+class D
+{
+    public int P2 { get; set; }
+}
+";
+            await VerifyNoItemsExistAsync(markup);
+        }
+
+        [Fact]
+        public async Task PropertiesInRecursivePattern_PositionalInFirstProperty_AfterCommaAndBeforeParen()
+        {
+            var markup =
+@"
+class Program
+{
+    public D P1 { get; set; }
+
+    void M()
+    {
+        _ = this is Program { P1: (1, $$) }
+    }
+}
+class D
+{
+    public int P2 { get; set; }
+}
+";
+            await VerifyNoItemsExistAsync(markup);
+        }
+
+        [Fact]
+        public async Task PropertiesInRecursivePattern_InPositional_Incomplete()
+        {
+            var markup =
+@"
+public class Program
+{
+    public int P1 { get; set; }
+
+    void M()
+    {
+        _ = this is ({ $$ }) // no deconstruction into 1 element
+    }
+
+    public void Deconstruct(out Program x, out Program y) => throw null;
+}
+";
+            // PROTOTYPE(patterns2): Like for typing arguments in methods, we should fall back to best overload resolution candidate
+            await VerifyNoItemsExistAsync(markup);
+        }
+
+        [Fact]
+        public async Task PropertiesInRecursivePattern_InPositional_Incomplete_WithTwoTypes()
+        {
+            var markup =
+@"
+public class Program
+{
+    void M()
+    {
+        _ = this is ({ $$ }) // no deconstruction into 1 element
```

Comment:
this is a shame because it dpeends on the rest of the contents being there after the {}.   What happen if all you've typed is  ```({ $$``?

Reply:
I think we should be able to make deconstruction works like method invocations, so that we use our best guess/overload while you're typing. Tracked by https://github.com/dotnet/roslyn/issues/26470

I'll add the test you suggested

Reply:
I think the question is: will this work if you only type `({ $$` without the closing brace and paren? is that even parsed as a pattern?

Reply:
It is [parsed as a pattern](https://sharplab.io/#v2:EYLgZgpghgLgrgJwgZwLQDk4BstWFiAJQkiQDsBjCAFQE8AHFAGhhCmRiYBMQBqAHwACABgAEggIwBuALAAoQQGZxAJlEBhUQG95oveOWCALKICyACgCU23frsB9UQF5RMABYBLZKK+jzW0Vs9AF95YKA===) (I suspect because what can follow an `is` is fairly restricted, so that guides the parsing).
But the semantic information is no better than when the closing brace and paren are present.